install/install.py

```diff
@@ -11,174 +11,198 @@
-        else:
-            Colorizer.normal("Skipping distro related installation :(")
-
-        if args.core_only:
-            return
-
-        Colorizer.normal("Upgrading pip to the latest version ...")
-        # Upgrade pip before install required libraries
-        self.run_command("sudo pip2 install --upgrade pip")
-        # mitigate cffi errors by upgrading it first
-        self.run_command("sudo pip2 install --upgrade cffi")
-        # run next block only when distro_num == 1
-        if distro_num == '1':
-            # check kali major release number 0.x, 1.x, 2.x
-            kali_version = os.popen("cat /etc/issue", "r").read().split(" ")[2][0]
-            if kali_version == '1':
-                if args.no_user_input:
-                    fixsetuptools = 'n'
-                else:
-                    # ask the user if they really want to delete the symlink
-                    fixsetuptools = raw_input("Delete /usr/lib/python2.7/dist-packages/setuptools.egg-info? (y/n)\n(recommended, solves some issues in Kali 1.xx)")
-
-                if fixsetuptools == 'y':
-                    # backup the original symlink
-                    Colorizer.normal("Backing up the original symlink...")
-                    ts = time.time()
-                    human_timestamp = datetime.fromtimestamp(ts).strftime('%Y-%m-%d-%H:%M:%S')
-
-                    # backup the original symlink
-                    symlink_orig_path = "/usr/lib/python2.7/dist-packages/setuptools.egg-info"
-                    self.run_command("mv %s %s-BACKUP-%s" % (symlink_orig_path, symlink_orig_path, human_timestamp))
-
-                    Colorizer.info("The original symlink exists at %s-BACKUP-%s" % (symlink_orig_path, human_timestamp))
-
-                    # Finally owtf python libraries installed using pip
-                    self.install_using_pip(self.owtf_pip)
+            Colorizer.info("[*] %s has been automatically detected... " % distro)
+            Colorizer.normal("[*] Continuing in auto-mode")
+            break
+
+        if args.no_user_input:
+            distro_num = 0
+            break
+
+        print("")
+        for i, item in enumerate(cp.sections()):
+            Colorizer.warning("(%d) %s" % (i + 1, item))
+        Colorizer.warning("(0) My distro is not listed :( %s" % distro)
+
+        distro_num = raw_input("Select a number based on your distribution : ")
+        try:
+            # Checking if valid input is received
+            distro_num = int(distro_num)
+            break
+        except ValueError:
+            print('')
+            Colorizer.warning("[!] Invalid Number specified")
+            continue
+
+    # First all distro independent stuff is installed
+    install_restricted_from_cfg(restricted_cfg)
+
+    if distro_num != 0:
+        run_command(cp.get(cp.sections()[int(distro_num)-1], "install"))
+    else:
+        Colorizer.normal("[*] Skipping distro related installation :(")
+
+    if args.core_only:
```

Comment:
Never happened with me, I can't find any declaration of `.core_only` attribute too :confused: . @DePierre a one-liner on this ?